packages/ethers/src/index.ts

```diff
@@ -145,31 +146,30 @@ export class TurnkeySigner extends ethers.Signer implements TypedDataSigner {
     // https://github.com/ethers-io/ethers.js/blob/f97b92bbb1bde22fcc44100af78d7f31602863ab/packages/wallet/src.ts/index.ts#L117-L121
     if (unsignedTx.from != null) {
       const selfAddress = await this.getAddress();
-      if (ethers.utils.getAddress(unsignedTx.from) !== selfAddress) {
+      if (getAddress(unsignedTx.from) !== selfAddress) {
         throw new Error(
           `Transaction \`tx.from\` address mismatch. Self address: ${selfAddress}; \`tx.from\` address: ${unsignedTx.from}`
         );
       }
 
       delete unsignedTx.from;
     }
-
-    const serializedTx = ethers.utils.serializeTransaction(
-      unsignedTx as UnsignedTransaction
-    );
+    
+    const serializedTx = Transaction.from(unsignedTx).unsignedSerialized
     const nonHexPrefixedSerializedTx = serializedTx.replace(/^0x/, "");
     const signedTx = await this._signTransactionWithErrorWrapping(
       nonHexPrefixedSerializedTx
     );
+
     return `0x${signedTx}`;
   }
 
   // Returns the signed prefixed-message. Per Ethers spec, this method treats:
   // - Bytes as a binary message
   // - string as a UTF8-message
   // i.e. "0x1234" is a SIX (6) byte string, NOT 2 bytes of data
-  async signMessage(message: string | Bytes): Promise<string> {
-    const hashedMessage = ethers.utils.hashMessage(message);
+  async signMessage(message: string | Uint8Array): Promise<string> {
```

Comment:
I was basing it on the `signMessage` method on the [AbstractSigner](https://github.com/ethers-io/ethers.js/blob/a26ff77c210660e641455150ccea6c0530f02f61/src.ts/providers/abstract-signer.ts#L239) class:
```
abstract signMessage(message: string | Uint8Array): Promise<string>;
```


but looks like `ByteLike` is equivalent? The only difference being in the nomenclature:
```
export type DataHexString = string;
/**
 *  A string which is prefixed with ``0x`` and followed by any number
 *  of case-agnostic hexadecimal characters.
 *
 *  It must match the regular expression ``/0x[0-9A-Fa-f]*\/``.
 */
export type HexString = string;
/**
 *  An object that can be used to represent binary data.
 */
export type BytesLike = DataHexString | Uint8Array;
```
([source](https://github.com/ethers-io/ethers.js/blob/a26ff77c210660e641455150ccea6c0530f02f61/lib.esm/utils/data.d.ts#L5-L16))

I am not partial to either, your call!

Reply:
ah totally makes sense! let's stick with what you have. 

but if we did want to use `BytesLike`, it includes `string`, so it would simply be `signMessage(message: BytesLike)`